scss/_navbar.scss

```diff
@@ -139,13 +139,49 @@
 // Keep as a separate element so folks can easily override it with another icon
 // or image file as needed.
 .navbar-toggler-icon {
+  position: relative;
   display: inline-block;
-  width: 1.5em;
-  height: 1.5em;
-  vertical-align: middle;
-  background-repeat: no-repeat;
-  background-position: center;
-  background-size: 100%;
+  width: $navbar-toggler-font-size;
+  height: $border-width * 2;
+  vertical-align: super;
+  background: currentColor;
+  @include transition(background .3s ease);
+
+  &::before,
+  &::after {
+    position: absolute;
+    left: 0;
+    width: inherit;
+    height: inherit;
+    content: "";
+    background: currentColor;
+    transform: rotate3d(0, 0, 1, 0);
+    transform-origin: 50% 50%;
+    @include transition(.3s ease);
+  }
+  &::before {
+    top: $navbar-toggler-padding-y * -2;
+  }
+  &::after {
+    top: $navbar-toggler-padding-y * 2;
+  }
```

Comment:
I'd introduce a new variable for this use case, to dissociate those offsets from the button's padding.